sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -522,10 +522,20 @@ namespace Azure { namespace Core { namespace Http { namespace _detail {
   }
   void WinHttpAction::CompleteActionWithError(DWORD_PTR stowedErrorInformation, DWORD stowedError)
   {
-    // Note that the order of scope_exit and lock is important - this ensures that scope_exit is
-    // destroyed *after* lock is destroyed, ensuring that the event is not set to the signalled
-    // state before the lock is released.
-    auto scope_exit{m_actionCompleteEvent.SetEvent_scope_exit()};
+    if (m_expectedStatus != WINHTTP_CALLBACK_STATUS_HANDLE_CLOSING)
+    {
+      // Note that the order of scope_exit and lock is important - this ensures that scope_exit is
+      // destroyed *after* lock is destroyed, ensuring that the event is not set to the signalled
+      // state before the lock is released.
+      auto scope_exit{m_actionCompleteEvent.SetEvent_scope_exit()};
```

Comment:
This means the m_actionCompleteEvent will never get set.  Won't that result in a leak?

Reply:
we don't want to set if the expected one is status closing, we get the status closing also

Reply:
NOte that this makes a fairly significant change in the behavior, one which is likely to cause problems.

The old version sets the event *after* the m_stowedErrorInformation and m_stowedError were set.
The new version sets the event *before* the m_stowedErrorInformation and m_stowedError are set.

Reply:
NOte that you could probably either wrap the entire function in the m_expectedStatus if, or you could move this line below the code which sets the stowed exception.